File: annotationprocessingtesting/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyBooleanPropertyElement.java
package com.tickaroo.tikxml.annotationprocessing.propertyelement;

import com.tickaroo.tikxml.annotation.PropertyElement;
import com.tickaroo.tikxml.annotation.Xml;

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
public class EmptyBooleanPropertyElement {
  @PropertyElement
  boolean empty;

  @Override public boolean equals(Object o) {
    if (this == o) return true;
    if (!(o instanceof EmptyBooleanPropertyElement)) return false;

    EmptyBooleanPropertyElement that = (EmptyBooleanPropertyElement) o;

    return empty == that.empty;
  }

  @Override public int hashCode() {
    return (empty ? 1 : 0);
  }
}

File: annotationprocessingtesting-kt/src/main/java/com/tickaroo/tikxml/annotationprocessing/propertyelement/EmptyBooleanPropertyElement.kt
package com.tickaroo.tikxml.annotationprocessing.propertyelement

import com.tickaroo.tikxml.annotation.PropertyElement
import com.tickaroo.tikxml.annotation.Xml

/**
 * @author Hannes Dorfmann
 */
@Xml(name = "emptyPropertyTag")
class EmptyBooleanPropertyElement {
    @PropertyElement
    var empty: Boolean = false

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is EmptyBooleanPropertyElement) return false

        val that = other as EmptyBooleanPropertyElement?

        return empty == that!!.empty
    }

    override fun hashCode(): Int {
        return if (empty) 1 else 0
    }
}
